client/src/main/java/com/alibaba/nacos/client/config/NacosConfigService.java

```diff
@@ -241,6 +247,13 @@ private boolean publishConfigInner(String tenant, String dataId, String group, S
             params.add("tag");
             params.add(tag);
         }
+        if (extendFields != null) {
+            Object type = extendFields.get("type");
+            if (type != null && StringUtils.isNotEmpty(String.valueOf(type))) {
```

Comment:
`type != null ` is not needed as Java will always choose the most suitable overloading method with the provided type(`Object`) and `null` is a value for this variable of type `Object`. You could try:

```
Object o = null;
String.valueOf(o);
```